lib/faraday/restrict_ip_addresses.rb

```diff
@@ -47,7 +47,7 @@ def call(env)
     end
 
     def denied?(env)
-      addresses(env[:url].host).any? { |a| denied_ip?(a) }
+      addresses(env[:url].hostname).any? { |a| denied_ip?(a) }
```

Comment:
This should be a safe change, as per [the documentation](https://ruby-doc.org/stdlib-2.1.1/libdoc/uri/rdoc/URI/Generic.html#method-i-hostname):

> This method is same as #host except brackets for IPv6 (andn future IP) addresses are removed.